portal-impl/src/com/liferay/portlet/internal/PortletBagImpl.java

```diff
@@ -184,16 +183,21 @@ public List<PreferencesValidator> getPreferencesValidatorInstances() {
 
 	@Override
 	public ResourceBundle getResourceBundle(Locale locale) {
+		Snapshot<ResourceBundleLoader> resourceBundleLoaderSnapshot =
```

Comment:
@JanisZhang The dynamic `Snapshot` should be an instance field, otherwise we're creating a service tracker every time we call this method.